### ISSUE TITLE
Reverse Engineer - Generate from sqlproj doesn't work

### ISSUE THREAD

**Issue by danielforwardcamp** [ErikEJ/EFCorePowerTools#1241]
Trying to reverse engineer from sqlproj doesn't work on Visual Studio 2022 with EF Net Core 6

### Steps to reproduce
- EF Core Power Tools -> Reverse Engineer
- Choose SQL Server Database Project (.dacpac) -> I select my sqlproj
- EF Core 6
- Ok
- In Output - Extensions: System.Exception: 
```
Dacpac build failed, please pick the file manually
   at async Task<string> EFCorePowerTools.Extensions.EnvDteExtensions.BuildSqlProjAsync(string sqlprojPath)
   at async Task<DatabaseConnectionModel> EFCorePowerTools.Handlers.ReverseEngineer.ReverseEngineerHandler.GetDatabaseInfoAsync(ReverseEngineerOptions options)
   at async Task EFCorePowerTools.Handlers.ReverseEngineer.ReverseEngineerHandler.ReverseEngineerCodeFirstAsync(Project project, string optionsPath, bool onlyGenerate)
```

If I select manually the .dacpac file generated by the .sqlproj project then:
```
System.Exception: Table list error: 
Microsoft.SqlServer.Dac.Model.DacModelException: Add or update objects failed due to the following errors: 
Error SQL46029: Unexpected end of file occurred.

   at void Microsoft.SqlServer.Dac.Model.SqlSchemaModelObjectService.ThrowIfModelErrorsExist(string errorMessage)
   at void Microsoft.SqlServer.Dac.Model.SqlSchemaModelObjectService.DoAddOrUpdateObjects(TSqlScript inputScriptAst, string sourceName, TSqlObjectOptions options)
   at void GOEddie.Dacpac.References.DacpacMerger.Merge() in C:/Code/EFCorePowerTools/src/GUI/ErikEJ.EntityFrameworkCore.SqlServer.Dacpac/DacpacConsolidate/DacpacMerger.cs:line 59
   at string GOEddie.Dacpac.References.DacpacConsolidator.Consolidate(string dacpacPath) in C:/Code/EFCorePowerTools/src/GUI/ErikEJ.EntityFrameworkCore.SqlServer.Dacpac/DacpacConsolidate/DacpacConsolidator.cs:line 23
   at DatabaseModel ErikEJ.EntityFrameworkCore.SqlServer.Scaffolding.SqlServerDacpacDatabaseModelFactory.Create(string connectionString, DatabaseModelFactoryOptions options) in C:/Code/EFCorePowerTools/src/GUI/ErikEJ.EntityFrameworkCore.SqlServer.Dacpac/Scaffolding/SqlServerDacpacDatabaseModelFactory.cs:line 59
   at List<DatabaseTable> RevEng.Core.TableListBuilder.GetTableDefinitions() in C:/Code/EFCorePowerTools/src/GUI/RevEng.Core/TableListBuilder.cs:line 67
   at List<TableModel> RevEng.Core.TableListBuilder.GetTableModels() in C:/Code/EFCorePowerTools/src/GUI/RevEng.Core/TableListBuilder.cs:line 40
   at int efreveng.Program.Main(string[] args) in C:/Code/EFCorePowerTools/src/GUI/efreveng/Program.cs:line 30

   at List<TableModel> EFCorePowerTools.Handlers.ReverseEngineer.ResultDeserializer.BuildTableResult(string output)
   at async Task<List<TableModel>> EFCorePowerTools.Handlers.ReverseEngineer.EfRevEngLauncher.GetTablesInternalAsync(string arguments)
   at async Task<List<TableModel>> EFCorePowerTools.Handlers.ReverseEngineer.EfRevEngLauncher.GetTablesAsync(string connectionString, DatabaseType databaseType, SchemaInfo[] schemas)
   at async Task<List<TableModel>> EFCorePowerTools.Handlers.ReverseEngineer.TableListBuilder.GetTableDefinitionsAsync(CodeGenerationMode codeGenerationMode)
   at async Task<List<TableModel>> EFCorePowerTools.Handlers.ReverseEngineer.ReverseEngineerHandler.GetDacpacTablesAsync(string dacpacPath, CodeGenerationMode codeGenerationMode)
   at async Task<bool> EFCorePowerTools.Handlers.ReverseEngineer.ReverseEngineerHandler.LoadDataBaseObjectsAsync(ReverseEngineerOptions options, DatabaseConnectionModel dbInfo, Tuple<List<Schema>, string> namingOptionsAndPath)
   at async Task EFCorePowerTools.Handlers.ReverseEngineer.ReverseEngineerHandler.ReverseEngineerCodeFirstAsync(Project project, string optionsPath, bool onlyGenerate)
```

EF Core version in use: EF Core 6

EF Core Power Tools version: 2.5.877.0

Database engine: SQL Server 2019

Visual Studio version: Visual Studio 2022 17.0.4

**Comment by danielforwardcamp** [comment]
I can't share my definition of dacpac. What kind of repro do you need?

**Comment by ErikEJ** [comment]
Without (private) access to your .sqlproj I am not able to assist. Apparently some file manipulation fails.

**Comment by danielforwardcamp** [comment]
@ErikEJ I already give you private access to the sqlproj. I'll wait for your response, thx.

**Comment by ErikEJ** [comment]
The tool attempts to merge all .dacpac into a single one, and for your project, merging either msdb.dacapac or master.dacpac fails. I have disabled this in the latest daily, and it seems to be able to work. Please try it out, and I will make it an option to not do dacpac merge if you can confirm the fix.

**Comment by danielforwardcamp** [comment]
With the latest build selecting from dacpac file is working now. So far so good. I can go with that option, but...

When selecting from sqlproj still having the same error.

The generated model has some errors:
- There are conflicts when the name of the Entity is equal to some name in System namespace (ie: entities with name: Action or Version)
- entity.HasOne method is missing the generic type, giving a compiling error: 
![image](https://user-images.githubusercontent.com/86418198/149299987-5614974f-f749-4d0a-9248-112aefb5aafe.png)

**Comment by ErikEJ** [comment]
Great news.

Does your project and the target EF version match?

You can use the renaming feature to fix the Version name issue?

**Comment by danielforwardcamp** [comment]
`Does your project and the target EF version match?`
Yes
.Net 6.0.1
EF Core 6.0.1
Target EF Core version (from EF Power Tools popup): EF Core 6

`You can use the renaming feature to fix the Version name issue?`
Could not be an option for me. We already are using the table names as entity names in large existing projects.

`When selecting from sqlproj still having the same error.`
Any news here?

Thx for the support @ErikEJ

**Comment by ErikEJ** [comment]
You are welcome, a sponsorship is always appreciated!

>> entity.HasOne method is missing the generic type, giving a compiling error

Is this a many to many entity? And are you disabling "hidden" many to many?

>> We already are using the table names as entity names in large existing projects.

How do you fix the name clash there?
Which part of the generated code does the error occur?

I will try to repro based on your shared project.

**Comment by danielforwardcamp** [comment]
> Is this a many to many entity? And are you disabling "hidden" many to many?

No, I won't generate the many to many entity in case is not required.
I used the default unchecked in "Use many to many entity (EF Core 6)" option
It's weird due to the entities where this happens are missing the Navigation Properties as well (CustomerTaxes and UsuarioDireccionEnvio), but the table's FK looks fine.

>How do you fix the name clash there?
>Which part of the generated code does the error occur?

We solve in two ways: giving an alias or using fully qualified name, but in these cases is our own managed code.
In this case is autogenerated so I guess is something in your side, right?
I think is happening if I overwrite the entity's namespace (when the namespace of the entity is not the same as the DbContext, in other case is using the right type)

**Comment by ErikEJ** [comment]
>> It's weird due to the entities where this happens are missing the Navigation Properties as well (CustomerTaxes and UsuarioDireccionEnvio), but the table's FK looks fine.

Could be a bug in the code that discovers the model from the dacpac, I will investigate

>> I think is happening if I overwrite the entity's namespace (when the namespace of the entity is not the same as the DbContext, in other case is using the right type)

I was not able to repro this when using the same namespace, I need a simple repro from you.

**Comment by ErikEJ** [comment]
Looks like the HasOne issue is related to many to many tables with EF 6 and .dacpac - most likely some bug.

**Comment by ErikEJ** [comment]
The HasOne issue looks like a EF Core issue, possibly will be fixed in 6.0.2 - in the meantime you can  check "Use many to many entity (EF Core 6)"

**Comment by ErikEJ** [comment]
Or if you can share a screen shot of the EF Core Power Tools advanced options (for the name clash issue)

**Comment by ErikEJ** [comment]
I was able to repro the name clash - simple fix is to remove: `using System;` manually

**Comment by danielforwardcamp** [comment]
> The HasOne issue looks like a EF Core issue, possibly will be fixed in 6.0.2 - in the meantime you can check "Use many to many entity (EF Core 6)"

Ok, we can wait for the next release

> I was able to repro the name clash - simple fix is to remove: using System; manually

I know that this will fix the problem, but there could be any possible solution from the tool? This is autogenerated code, so I guess I shouldn't modify manually since it will not stick around, right?

Thx @ErikEJ I think we can live with that

**Comment by ErikEJ** [comment]
You are very welcome, reopening to round out the implementation of making dacpac merge optional
